### PR TITLE
Enhancing the htx_block_device.py to support rpm based install

### PR DESCRIPTION
**Pull request by Naresh-ibm**
modified the script so that it support the rpm based install of htx tool and some other script enhancements.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[root@ltcden4-lp1 disk]# avocado run --test-runner runner htx_block_devices.py -m htx_block_devices.py.data/htx_block_devices.yaml
JOB ID     : fc02b53cfe3845852e3ac92f6c20e178efc89eeb
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-11-09T23.50-fc02b53/job.log
 (1/3) htx_block_devices.py:HtxTest.test_start;run-3a99: PASS (33.94 s)
 (2/3) htx_block_devices.py:HtxTest.test_check;run-3a99: PASS (60.96 s)
 (3/3) htx_block_devices.py:HtxTest.test_stop;run-3a99: PASS (10.98 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-11-09T23.50-fc02b53/results.html
JOB TIME   : 116.19 s
[root@ltcden4-lp1 disk]#

**Comment by Naresh-ibm** [comment]
[htx_block_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9978438/htx_block_job.log)

File: io/disk/htx_block_devices.py [hunk 1]
```diff
@@ -68,38 +72,86 @@ def setup_htx(self):
...
-        if detected_distro in ['centos', 'fedora', 'rhel', 'redhat']:
+        self.detected_distro = distro.detect()
+        if self.detected_distro.name in ['centos', 'fedora', 'rhel', 'redhat']:
             packages.extend(['gcc-c++', 'ncurses-devel', 'tar'])
-        elif detected_distro == "Ubuntu":
+        elif self.detected_distro.name == "Ubuntu":
             packages.extend(['libncurses5', 'g++', 'ncurses-dev',
                              'libncurses-dev', 'tar'])
-        elif detected_distro == 'SuSE':
+        elif self.detected_distro.name == 'SuSE':
```
Comment (abdhaleegit) [inline io/disk/htx_block_devices.py:81]: self.detected_distro.name  calling .name 4 times... instead earlier `detected_distro = distro.detect().name`   was called one time.. and the param can be sued in 4 places.. any reason to make it multiple lines ?

File: io/disk/htx_block_devices.py [hunk 2]
```diff
@@ -68,38 +72,86 @@ def setup_htx(self):
...
-        if process.system('./installer.sh -f'):
-            self.fail("Installation of htx fails:please refer job.log")
+        if self.run_type == 'git':
+            url = "https://github.com/open-power/HTX/archive/master.zip"
+            tarball = self.fetch_asset("htx.zip", locations=[url], expire='7d')
+            archive.extract(tarball, self.teststmpdir)
+            htx_path = os.path.join(self.teststmpdir, "HTX-master")
+            os.chdir(htx_path)
+
+            exercisers = ["hxecapi_afu_dir", "hxedapl", "hxecapi", "hxeocapi"]
```
Comment (abdhaleegit) [inline io/disk/htx_block_devices.py:98]: can we have exercisers from yaml param.. so we can always change

File: io/disk/htx_block_devices.py [hunk 3]
```diff
@@ -217,6 +269,7 @@ def stop_htx(self):
             cmd = "htxcmdline -shutdown -mdt %s" % self.mdt_file
             process.system(cmd, timeout=120, ignore_status=True)
 
-        daemon_state = process.system_output('/etc/init.d/htx.d status')
-        if daemon_state.decode("utf-8").split(" ")[-1] == 'running':
-            process.system('/usr/lpp/htx/etc/scripts/htxd_shutdown')
+        #cmd = '/usr/lpp/htx/etc/scripts/htx.d status'
+        #daemon_state = process.system_output(cmd)
+        #if daemon_state.decode("utf-8").split(" ")[-1] == 'running':
+        #    process.system('/usr/lpp/htx/etc/scripts/htxd_shutdown')
```
Comment (abdhaleegit) [inline io/disk/htx_block_devices.py:275]: any reason to commentt out all lines ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: io/disk/htx_block_devices.py (hunk 2 above)
Comment (Naresh-ibm) [reply io/disk/htx_block_devices.py:98]: @abdhaleegit This is something to do internally and not an input here.
So I think having this internally is ok I feel.

File: io/disk/htx_block_devices.py (hunk 3 above)
Comment (Naresh-ibm) [reply io/disk/htx_block_devices.py:275]: forgot to remove while pushing.
will be removing this.

**Comment by Naresh-ibm** [comment]
SLES Logs
-----------------------
FC/vFC
------------
localhost:~ # avocado run --test-runner runner htx_block_devices.py -m htx_block_devices.yaml
JOB ID     : 24d08002826ba8f190c531467cbecd4cb2640855
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-11-28T12.00-24d0800/job.log
 (1/3) htx_block_devices.py:HtxTest.test_start;run-5fbb: PASS (69.70 s)
 (2/3) htx_block_devices.py:HtxTest.test_check;run-5fbb: PASS (61.09 s)
 (3/3) htx_block_devices.py:HtxTest.test_stop;run-5fbb: PASS (12.10 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-11-28T12.00-24d0800/results.html
JOB TIME   : 155.64 s
localhost:~ #


nvme/nvmf
-------------------

localhost:~ # avocado run --test-runner runner htx_block_devices.py -m htx_block_devices.yaml
JOB ID     : 5f0d43b15a1553b3966e171087fa5b84ca3eb5b8
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-11-28T11.54-5f0d43b/job.log
 (1/3) htx_block_devices.py:HtxTest.test_start;run-eca4: PASS (69.59 s)
 (2/3) htx_block_devices.py:HtxTest.test_check;run-eca4: PASS (61.25 s)
 (3/3) htx_block_devices.py:HtxTest.test_stop;run-eca4: PASS (12.21 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-11-28T11.54-5f0d43b/results.html
JOB TIME   : 153.11 s
localhost:~ #

**Comment by Naresh-ibm** [comment]
[sles_FC_vFC_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/10101896/sles_FC_vFC_job.log)
[sles_nvme_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/10101897/sles_nvme_job.log)

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**